src/pages/resultsView/plots/PlotsTab.tsx

```diff
@@ -689,13 +694,175 @@ export default class PlotsTab extends React.Component<IPlotsTabProps, {}> {
             set selectedTreatmentOption(o: any) {
                 this._selectedTreatmentOption = o;
             },
-            _selectedGeneOption: undefined,
-            _selectedGenesetOption: undefined,
-            _selectedTreatmentOption: undefined,
-            _dataType: undefined,
-            _mutationCountBy: undefined,
-            _logScale: false,
-            _selectedDataSourceOption: undefined,
+
+            get _selectedGeneOption() {
+                const param = vertical
+                    ? self.props.urlWrapper.query.plots_selectedGeneOption_vert
+                    : self.props.urlWrapper.query.plots_selectedGeneOption_horz;
+
+                if (!param) {
+                    return undefined;
+                } else {
+                    const val = parseFloat(param);
+                    const geneOptions = vertical
+                        ? self.vertGeneOptions
+                        : self.horzGeneOptions.result || [];
+
+                    return geneOptions.find(o => o.value === val);
+                }
+            },
+            set _selectedGeneOption(o: any) {
+                const newParams: Mutable<Partial<ResultsViewURLQuery>> = {};
+                if (vertical) {
+                    newParams.plots_selectedGeneOption_vert = o && o.value;
+                } else {
+                    newParams.plots_selectedGeneOption_horz = o && o.value;
+                }
+                self.props.urlWrapper.updateURL(newParams);
+            },
+
+            get _selectedGenesetOption() {
+                const optionVal = vertical
+                    ? self.props.urlWrapper.query
+                          .plots_selectedGenesetOption_vert
+                    : self.props.urlWrapper.query
+                          .plots_selectedGenesetOption_horz;
+
+                if (!optionVal) {
+                    return undefined;
+                } else {
+                    const genesetOptions =
+                        (vertical
+                            ? self.vertGenesetOptions
+                            : self.horzGenesetOptions.result) || [];
+
+                    return genesetOptions.find(o => o.value === optionVal);
+                }
+            },
+            set _selectedGenesetOption(o: any) {
+                const newParams: Mutable<Partial<ResultsViewURLQuery>> = {};
```

Comment:
What does Mutable do?

Reply:
for some reason Partial<ResultsViewURLQuery> was coming out with readonly props. Mutable just makes it not readonly